### ISSUE TITLE
2019.x Android package: Install SDK & NDK Tools

### ISSUE THREAD

**Issue by sp-ricard-valverde** [DragonBox/u3d#359]
### Issue Checklist

- [x] I'm using the latest u3d version
- [x] I ran `u3d --help`
- [x] I read the [README](https://github.com/DragonBox/u3d/blob/master/README.md)
- [x] I made sure that a similar [issue](https://github.com/DragonBox/u3d/issues) doesn't exit

### Issue Description

Starting from Unity 2019, the Unity Hub offers the possibility to install 'Android SDK & NDK Tools' along with the Android Build Support package. This makes the provisioning process much more easier as there's no need to open the Editor and download those packages separately.

When listing the packages for Unity 2019(u3d available -p) versions there's no SDK & NDK available, not are they installed when installing the 'Android' package along the Editor(u3d install 2019 -p Unity,Android).

It would be great to have this as u3d could be enough to provision an Android Unity build machine for CI environments.

**Comment by sp-ricard-valverde** [comment]
Furthermore, the only way to install the supported Android SDK & NDK is from the Hub at Unity installation time from version 2019 onwards. 
If the Unity version is not installed from the Hub there's no apparent way of knowing which SDK & NDK it supports(it's causing strange Editor crashes when mixing SDK & NDK versions for us).

**Comment by bdominguez** [comment]
Same problem here... As you also said they hide the specfic NDK version so we can't install it manually neither...

**Comment by phil-flyclops** [comment]
Adding a reference image of the available packages for a 2019 install.  It ncludes android but does not include the nested android packages.
![Screen Shot 2019-09-10 at 10 00 42 AM](https://user-images.githubusercontent.com/44981522/64620511-000cf780-d3b2-11e9-8b17-2cd7e3e49766.png)

**Comment by sp-ricard-valverde** [comment]
You can now at least install those packages using UnityHub CLI, although the tool is very lacking and error prone if trying to install multiple packages at a time.

**Comment by umegaya** [comment]
@sp-ricard-valverde 

> You can now at least install those packages using UnityHub CLI

could you elaborate this? do you means we can run /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub from command line like UnityEditor?

**Comment by phil-flyclops** [comment]
I'm not sure about the CLI side of things, but if you use UnityHub UI to install the packages you can get similar run ability of different versions of editors to u3d by doing the following, although it misses out on some nice log culling features in u3d

```
VERSION=`cat ProjectSettings/ProjectVersion.txt | awk '{print $2}' | head -1`
UNITY_PATH="/Applications/Unity/Hub/Editor/${VERSION}/Unity.app/Contents/MacOS/Unity"

# u3d run less-than-ideal-because-of-logging replacement
$UNITY_PATH -batchmode ...
```

**Comment by sp-ricard-valverde** [comment]
> @sp-ricard-valverde
> 
> > You can now at least install those packages using UnityHub CLI
> 
> could you elaborate this? do you means we can run /Applications/Unity\ Hub.app/Contents/MacOS/Unity\ Hub from command line like UnityEditor?

Unity Hub introduced the ability to install Unity packages from command line(including ndk and skd), although it's a feature in preview, it's fully working:

https://unity3d.com/hub/whats-new

See version 2.1.0 notes.

**Comment by umegaya** [comment]
@sp-ricard-valverde oic thanks! I missed the update and temporary using https://github.com/sttz/install-unity, which can be installed via homebrew, but it would be nicer if official CLI installer is provided 👍

**Comment by bdominguez** [comment]
@lacostej @niezbop 

Before Unity showed a button on editor preferences to download the exact version of the NDK, now since it allows you to install from the hub the NDK does not even appear which version to use if you want to download it on your own.

That's why it's very important that from u3d the NDK module appears as downloadable and installable so we can get the expected version.

**Comment by niezbop** [comment]
Hey @bdominguez!

We agree that it's an important feature, and I'm working on it right now 😄 See https://github.com/niezbop/u3d/tree/feature/installer/sdk_ndk 

This branch should expand upon the existing modules listing by using the ones the hub offer, including the SDK and NDK plus their tool.

It's still WIP, but I should be able to open a PR and iterate upon it. This will likely be the last feature that we implement before rolling out the next release, so this should come in your hands in the very near future!

In the mean time, [Unity documents which NDK you should be running](https://docs.unity3d.com/Manual/android-sdksetup.html) in case you didn't know.

**Comment by bdominguez** [comment]
Thanks @niezbop I didn't know!

**Comment by niezbop** [comment]
Hey everyone!

Just to keep you up-to-date, the feature is nearing completion!
```
u3d (feature/installer/sdk_ndk) $ u3d install 2019.2.12f1 -p android-ndk
Unity 2019.2.12f1 is already installed
Root privileges are required
Password for paulniezborala:
**********************
No credentials storage available
Android-ndk is already downloaded
--------------------------------------------
--- Installing Android NDK (android-ndk) ---
--------------------------------------------
Installing with /Users/paulniezborala/Downloads/Unity_Packages/2019.2.12f1/android-ndk-r16b-darwin-x86_64.zip
$ chmod a+w /Applications/Unity_2019.2.12f1/PlaybackEngines/AndroidPlayer
Successfully unizpped android-ndk-r16b-darwin-x86_64.zip at /Applications/Unity_2019.2.12f1/PlaybackEngines/AndroidPlayer/NDK
```

There are of courses a few things to tweak still, and we have to iterate just a bit on a it yet, but if you want to give it a go, and try if it works on your machine you can build it locally (clone the repo and run ./build.sh). Any feedback appreciated!

**Comment by bdominguez** [comment]
Great job @niezbop

**Comment by niezbop** [comment]
Hey all.

This has been released as part of [u3d 1.2.0](https://rubygems.org/gems/u3d/versions/1.2.0)! Thanks once again for your patience on this!
